updater.go

```diff
@@ -203,3 +174,41 @@ func applyUpdate(r io.Reader, publicKey ed25519.PublicKey, signature [64]byte) (
 	}
 	return opts.TargetPath, nil
 }
+
+func triggerSchedule(updater *Updater) {
+	for {
+		var next time.Duration
+
+		if updater.conf.Schedule.Interval != 0 {
+			next = updater.conf.Schedule.Interval
+		}
+		if updater.conf.Schedule.At.Repeating != None {
+			at := nextScheduleAt(updater.conf.Schedule.At.Repeating, updater.conf.Schedule.At.Time)
+			if next == 0 || at < next {
+				next = at
+			}
+		}
+
+		time.Sleep(next)
+		logInfo("Scheduled upgrade check after %s.\n", next)
+		err := updater.CheckNow()
+		if err != nil {
+			logError("Upgrade error: %v\n", err)
+		}
+	}
+}
+
+func nextScheduleAt(repeating Repeating, offset time.Time) time.Duration {
```

Comment:
Much like the comment above this name implies returning a time but it returns a duration.
Perhaps "delayUntilNextTrigger" or "offsetToSchedule" etc?